nuklear_console_row.h

```diff
@@ -161,57 +140,67 @@ NK_API struct nk_rect nk_console_row_render(nk_console* console) {
   }
 
   // Consume mouse movement before children have a chance to.
-  int numChildren = (int)cvector_size(row->data.children);
+  int numChildren = (int)cvector_size(console->children);
   struct nk_input* input = &console->context->input;
   if (console->selectable && top->input_processed == nk_false &&
       widget_bounds.w > 0 && nk_input_is_mouse_moved(input) &&
       nk_input_is_mouse_hovering_rect(input, widget_bounds)) {
     nk_console_set_active_widget(console);
-    // Calculate the active child index based on where the mouse is within
-    // widget_bounds. This assumes each column has equal width.
-    float columnWidth = (float)widget_bounds.w / console->columns;
-    float mouseX = input->mouse.pos.x - widget_bounds.x;
-    row->data.activeChild = 0;
-    for (float x = 0; x < widget_bounds.w;) {
-      float right =
-          x + row->data.children[row->data.activeChild].columns * columnWidth;
-      if (right > mouseX) {
-        break;
-      }
-      x = right;
-      row->data.activeChild += 1;
+
+    // Find the child that the mouse is hovering over.
+    if (console->columns > 0) {
+        float x_percent = (input->mouse.pos.x - widget_bounds.x) / widget_bounds.w;
+        float column_width_percent = 0.0f;
+        for (int i = 0; i < numChildren; i++) {
+            int widget_columns = console->children[i]->columns;
+            if (widget_columns <= 0) {
+                widget_columns = 1;
+            }
+            column_width_percent += (float)widget_columns / (float)console->columns;
+            if (x_percent < column_width_percent) {
+                data->activeChild = i;
+                break;
+            }
+        }
+    }
```

Comment:
This updated loop gets the active widget based on the percent of maximum columns in the row.

Reply:
Looks great!